cmd/diff.go

```diff
@@ -80,24 +79,28 @@ func runDiff(cmd *cobra.Command, _ []string) error {
 		return nil
 	}
 
-	// Create appropriate source based on kustomize flag
+	// Create appropriate source based on ref type
 	var source manifest.Source
-	if kustomizeEnabled {
-		// Validate that ref is a directory
-		if finfo, ferr := os.Stat(ref); ferr != nil || !finfo.IsDir() {
-			refErr := fmt.Errorf("with --kustomize flag, --ref must be a valid directory path: %s", ref)
-			log.Error(refErr.Error())
-			return refErr
-		}
+
+	isValidDirectoryRef := utils.IsValidDirectoryRef(ref)
+	// If kustomize is enabled, ref must be a valid directory
+	if kustomizeEnabled && !isValidDirectoryRef {
+		refErr := fmt.Errorf("with --kustomize flag, --ref must be a valid directory path: %s", ref)
+		log.Error(refErr.Error())
```

Comment:
We should return the error as a structured way in order to be able to parse it with queries.
```suggestion
		log.Error("missing configuration", "error", refErr)
```